### DISCUSSION TITLE
use of FT4232H FTDI USB Serial Device converter

### DISCUSSION THREAD

**Discussion by rockoguido** [polarfire-soc/polarfire-soc-documentation#283] [Bare metal embedded software] [answered]
Have an eval board produced by a 3rd party Sundance, 
 link to the baseboard is
https://www.sundance.technology/som-cariers/pci-express-fmc-carriers/solar-express-200/

On this board is a SOM based on a MPF500T
https://www.sundancedsp.com/som/polarfire/som-pf1-fpga-module-with-mpf500-1fcg1152/

This Sundance platform uses the same FTDI USB Serial Device Converter (FT4232H) to manage JTAG and UART based connectivity between host and target as does the Polarfire Eval Kit from Microchip

Essentially Sundance copied the design from our Eval card and that’s fine.    

Using the miv-rv32i-systick-blinky app to validate a working UART on the platform.  

For point of reference also using the Microchip Polar Eval Kit (based on MPF300T)

Both platforms have a valid bitstream and when connected to a windows host the JTAG and  UART works as expected.   

The issue is when we connect from the Sundance platform to a Linux host (Unbuntu 20.04)
We can connect via JTAG ok and execute the application but the UART I/O does not come out on the host port via either putty or minicom
The details are as follows...

In all cases using the same app (miv-rv32i-systick-blnky)

Eval card connected to Windoze host
-	4 TTY devices instantiated on the host and both JTAG and UART connectivity work as expected

SE200 & SOM1 connected to Windoze host
-	4 TTY devices instantiated on the host and both JTAG and UART connectivity work as expected

Eval card connected to Linux host (Ubuntu 20.04)
-	1 TTY device instantiated (/dev/ttyUSB0) and both JTAG and UART connectivity work as expected

SE200 & SOM1 connected to Linux host (Ubuntu 20.04)
-	1 TTY device instantiated (/dev/ttyUSB0) and JTAG is operational but don’t see any UART related activity
Don't see this as a hardware issue since the only difference between working and non working is the host platform but..  I'm just a software guy..  

On the Linux side I've verified the instantiation of /dev/ttyUSB0 and Linux recognizing the FlashPro5 and FTDI USB Serial Device FT4232H

any input greatly appreciated!
Thx
Dave

**Comment by hughbreslin** [comment, marked as answer]
Hi @rockoguido should this be on the [Mi-V Soft CPUs Discussions](https://github.com/orgs/Mi-V-Soft-RISC-V/discussions)?

**Reply by rockoguido** [reply]
just moved it to [Mi-V Soft RISC-V] discussion

**Reply by hughbreslin** [reply]
Thanks :)
